arch/x86/adler32_avx512.c

```diff
@@ -15,13 +15,37 @@
 
 #ifdef X86_AVX512_ADLER32
 
+#ifdef X86_AVX2_ADLER32 
```

Comment:
Have you thought about just including "cpu_features.h"?

Reply:
I could, I figured it made since to limit the scope of things pulled in, but it wouldn't be that much extra work on the compiler.

I think I found the issue btw, though it's strange.  It stems from us extending the xmm register content to a 256 bit wide register.  From what I've read that's supposed to zero extend, however from the looks of it the upper half may be undefined.  I can fix this with a mask of sorts, but I'll see what I can do to make sure this does what it's supposed to.

Reply:
If you want to limit it in scope you could put the externs in adler32_p.h instead. What do you think?

Reply:
Well adler32_p.h is multi-arch so I don't know..